==38938== Helgrind, a thread error detector
==38938== Copyright (C) 2007-2017, and GNU GPL'd, by OpenWorks LLP et al.
==38938== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==38938== Command: ./bin/memoria ./configs/fibo.config
==38938== Parent PID: 27221
==38938== 
==38938== ---Thread-Announcement------------------------------------------
==38938== 
==38938== Thread #2 was created
==38938==    at 0x49A39F3: clone (clone.S:76)
==38938==    by 0x49A48EE: __clone_internal (clone-internal.c:83)
==38938==    by 0x49126D8: create_thread (pthread_create.c:295)
==38938==    by 0x49131FF: pthread_create@@GLIBC_2.34 (pthread_create.c:828)
==38938==    by 0x4853767: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10E3F9: hilos_memoria (server.c:282)
==38938==    by 0x10C8A8: main (main.c:45)
==38938== 
==38938== ---Thread-Announcement------------------------------------------
==38938== 
==38938== Thread #1 is the program's root thread
==38938== 
==38938== ----------------------------------------------------------------
==38938== 
==38938==  Lock at 0x1181A0 was first observed
==38938==    at 0x4850CCF: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10C7C7: main (main.c:28)
==38938==  Address 0x1181a0 is 0 bytes inside data symbol "mutex_logs"
==38938== 
==38938== Possible data race during write of size 1 at 0x4AAA4A0 by thread #2
==38938== Locks held: none
==38938==    at 0x48597B4: mempcpy (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4909664: _IO_new_file_xsputn (fileops.c:1235)
==38938==    by 0x4909664: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==38938==    by 0x48FDA9C: fputs (iofputs.c:38)
==38938==    by 0x48767C6: txt_write_in_file (in /usr/lib/libcommons.so)
==38938==    by 0x4877522: _log_write_in_level (in /usr/lib/libcommons.so)
==38938==    by 0x48771D7: log_info (in /usr/lib/libcommons.so)
==38938==    by 0x111FC6: crear_conexion (sockets.c:163)
==38938==    by 0x10E21D: cliente_memoria_filesystem (server.c:218)
==38938==    by 0x10E32E: funcion_hilo_cliente (server.c:252)
==38938==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4912AC2: start_thread (pthread_create.c:442)
==38938==    by 0x49A3A03: clone (clone.S:100)
==38938== 
==38938== This conflicts with a previous write of size 1 by thread #1
==38938== Locks held: 1, at address 0x1181A0
==38938==    at 0x48597B4: mempcpy (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4909664: _IO_new_file_xsputn (fileops.c:1235)
==38938==    by 0x4909664: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==38938==    by 0x48FDA9C: fputs (iofputs.c:38)
==38938==    by 0x48767C6: txt_write_in_file (in /usr/lib/libcommons.so)
==38938==    by 0x4877522: _log_write_in_level (in /usr/lib/libcommons.so)
==38938==    by 0x48771D7: log_info (in /usr/lib/libcommons.so)
==38938==    by 0x10E47B: hilos_memoria (server.c:294)
==38938==    by 0x10C8A8: main (main.c:45)
==38938==  Address 0x4aaa4a0 is 0 bytes inside a block of size 4,096 alloc'd
==38938==    at 0x484A919: malloc (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x48FCBA3: _IO_file_doallocate (filedoalloc.c:101)
==38938==    by 0x490BCDF: _IO_doallocbuf (genops.c:347)
==38938==    by 0x490AF5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:744)
==38938==    by 0x49096D4: _IO_new_file_xsputn (fileops.c:1243)
==38938==    by 0x49096D4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==38938==    by 0x48FDA9C: fputs (iofputs.c:38)
==38938==    by 0x48767C6: txt_write_in_file (in /usr/lib/libcommons.so)
==38938==    by 0x4877522: _log_write_in_level (in /usr/lib/libcommons.so)
==38938==    by 0x48771D7: log_info (in /usr/lib/libcommons.so)
==38938==    by 0x10C801: main (main.c:29)
==38938==  Block was alloc'd by thread #1
==38938== 
==38938== ----------------------------------------------------------------
==38938== 
==38938==  Lock at 0x1181A0 was first observed
==38938==    at 0x4850CCF: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10C7C7: main (main.c:28)
==38938==  Address 0x1181a0 is 0 bytes inside data symbol "mutex_logs"
==38938== 
==38938== Possible data race during write of size 1 at 0x4AAC700 by thread #2
==38938== Locks held: none
==38938==    at 0x48597B4: mempcpy (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4909664: _IO_new_file_xsputn (fileops.c:1235)
==38938==    by 0x4909664: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==38938==    by 0x48F500B: outstring_func (vfprintf-internal.c:239)
==38938==    by 0x48F500B: __vfprintf_internal (vfprintf-internal.c:1517)
==38938==    by 0x48DE79E: printf (printf.c:33)
==38938==    by 0x4876800: txt_write_in_stdout (in /usr/lib/libcommons.so)
==38938==    by 0x4877570: _log_write_in_level (in /usr/lib/libcommons.so)
==38938==    by 0x48771D7: log_info (in /usr/lib/libcommons.so)
==38938==    by 0x111FC6: crear_conexion (sockets.c:163)
==38938==    by 0x10E21D: cliente_memoria_filesystem (server.c:218)
==38938==    by 0x10E32E: funcion_hilo_cliente (server.c:252)
==38938==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4912AC2: start_thread (pthread_create.c:442)
==38938== 
==38938== This conflicts with a previous write of size 1 by thread #1
==38938== Locks held: 1, at address 0x1181A0
==38938==    at 0x48597B4: mempcpy (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4909664: _IO_new_file_xsputn (fileops.c:1235)
==38938==    by 0x4909664: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==38938==    by 0x48F500B: outstring_func (vfprintf-internal.c:239)
==38938==    by 0x48F500B: __vfprintf_internal (vfprintf-internal.c:1517)
==38938==    by 0x48DE79E: printf (printf.c:33)
==38938==    by 0x4876800: txt_write_in_stdout (in /usr/lib/libcommons.so)
==38938==    by 0x4877570: _log_write_in_level (in /usr/lib/libcommons.so)
==38938==    by 0x48771D7: log_info (in /usr/lib/libcommons.so)
==38938==    by 0x10E47B: hilos_memoria (server.c:294)
==38938==  Address 0x4aac700 is 0 bytes inside a block of size 1,024 alloc'd
==38938==    at 0x484A919: malloc (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x48FCBA3: _IO_file_doallocate (filedoalloc.c:101)
==38938==    by 0x490BCDF: _IO_doallocbuf (genops.c:347)
==38938==    by 0x490AF5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:744)
==38938==    by 0x49096D4: _IO_new_file_xsputn (fileops.c:1243)
==38938==    by 0x49096D4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==38938==    by 0x48F500B: outstring_func (vfprintf-internal.c:239)
==38938==    by 0x48F500B: __vfprintf_internal (vfprintf-internal.c:1517)
==38938==    by 0x48DE79E: printf (printf.c:33)
==38938==    by 0x4876800: txt_write_in_stdout (in /usr/lib/libcommons.so)
==38938==    by 0x4877570: _log_write_in_level (in /usr/lib/libcommons.so)
==38938==    by 0x48771D7: log_info (in /usr/lib/libcommons.so)
==38938==    by 0x10C801: main (main.c:29)
==38938==  Block was alloc'd by thread #1
==38938== 
==38938== ---Thread-Announcement------------------------------------------
==38938== 
==38938== Thread #8 was created
==38938==    at 0x49A39F3: clone (clone.S:76)
==38938==    by 0x49A48EE: __clone_internal (clone-internal.c:83)
==38938==    by 0x49126D8: create_thread (pthread_create.c:295)
==38938==    by 0x49131FF: pthread_create@@GLIBC_2.34 (pthread_create.c:828)
==38938==    by 0x4853767: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10CA37: hilo_recibe_cpu (main.c:84)
==38938==    by 0x10C8D4: main (main.c:50)
==38938== 
==38938== ---Thread-Announcement------------------------------------------
==38938== 
==38938== Thread #60 was created
==38938==    at 0x49A39F3: clone (clone.S:76)
==38938==    by 0x49A48EE: __clone_internal (clone-internal.c:83)
==38938==    by 0x49126D8: create_thread (pthread_create.c:295)
==38938==    by 0x49131FF: pthread_create@@GLIBC_2.34 (pthread_create.c:828)
==38938==    by 0x4853767: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10DE6D: gestor_clientes (server.c:110)
==38938==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4912AC2: start_thread (pthread_create.c:442)
==38938==    by 0x49A3A03: clone (clone.S:100)
==38938== 
==38938== ----------------------------------------------------------------
==38938== 
==38938==  Lock at 0x1181A0 was first observed
==38938==    at 0x4850CCF: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10C7C7: main (main.c:28)
==38938==  Address 0x1181a0 is 0 bytes inside data symbol "mutex_logs"
==38938== 
==38938==  Lock at 0x1181E0 was first observed
==38938==    at 0x4854BFE: pthread_mutex_init (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10D6B4: inicializar_mutex (auxiliaresMem.c:67)
==38938==    by 0x10C829: main (main.c:33)
==38938==  Address 0x1181e0 is 0 bytes inside data symbol "mutex_lista_contextos_pids"
==38938== 
==38938== Possible data race during write of size 1 at 0x4AAC739 by thread #8
==38938== Locks held: 2, at addresses 0x1181A0 0x1181E0
==38938==    at 0x4859796: mempcpy (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4909664: _IO_new_file_xsputn (fileops.c:1235)
==38938==    by 0x4909664: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==38938==    by 0x48F500B: outstring_func (vfprintf-internal.c:239)
==38938==    by 0x48F500B: __vfprintf_internal (vfprintf-internal.c:1517)
==38938==    by 0x48DE79E: printf (printf.c:33)
==38938==    by 0x4876800: txt_write_in_stdout (in /usr/lib/libcommons.so)
==38938==    by 0x4877570: _log_write_in_level (in /usr/lib/libcommons.so)
==38938==    by 0x48771D7: log_info (in /usr/lib/libcommons.so)
==38938==    by 0x10EAEB: recibir_cpu (commCPU.c:98)
==38938==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4912AC2: start_thread (pthread_create.c:442)
==38938==    by 0x49A3A03: clone (clone.S:100)
==38938== 
==38938== This conflicts with a previous write of size 1 by thread #60
==38938== Locks held: none
==38938==    at 0x48597B4: mempcpy (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4909664: _IO_new_file_xsputn (fileops.c:1235)
==38938==    by 0x4909664: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==38938==    by 0x48F500B: outstring_func (vfprintf-internal.c:239)
==38938==    by 0x48F500B: __vfprintf_internal (vfprintf-internal.c:1517)
==38938==    by 0x48DE79E: printf (printf.c:33)
==38938==    by 0x4876800: txt_write_in_stdout (in /usr/lib/libcommons.so)
==38938==    by 0x4877570: _log_write_in_level (in /usr/lib/libcommons.so)
==38938==    by 0x48771D7: log_info (in /usr/lib/libcommons.so)
==38938==    by 0x111E90: cliente_handshake (sockets.c:127)
==38938==  Address 0x4aac739 is 57 bytes inside a block of size 1,024 alloc'd
==38938==    at 0x484A919: malloc (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x48FCBA3: _IO_file_doallocate (filedoalloc.c:101)
==38938==    by 0x490BCDF: _IO_doallocbuf (genops.c:347)
==38938==    by 0x490AF5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:744)
==38938==    by 0x49096D4: _IO_new_file_xsputn (fileops.c:1243)
==38938==    by 0x49096D4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==38938==    by 0x48F500B: outstring_func (vfprintf-internal.c:239)
==38938==    by 0x48F500B: __vfprintf_internal (vfprintf-internal.c:1517)
==38938==    by 0x48DE79E: printf (printf.c:33)
==38938==    by 0x4876800: txt_write_in_stdout (in /usr/lib/libcommons.so)
==38938==    by 0x4877570: _log_write_in_level (in /usr/lib/libcommons.so)
==38938==    by 0x48771D7: log_info (in /usr/lib/libcommons.so)
==38938==    by 0x10C801: main (main.c:29)
==38938==  Block was alloc'd by thread #1
==38938== 
==38938== ---Thread-Announcement------------------------------------------
==38938== 
==38938== Thread #4 was created
==38938==    at 0x49A39F3: clone (clone.S:76)
==38938==    by 0x49A48EE: __clone_internal (clone-internal.c:83)
==38938==    by 0x49126D8: create_thread (pthread_create.c:295)
==38938==    by 0x49131FF: pthread_create@@GLIBC_2.34 (pthread_create.c:828)
==38938==    by 0x4853767: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10E06E: servidor_memoria (server.c:166)
==38938==    by 0x10E11A: funcion_hilo_servidor (server.c:187)
==38938==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4912AC2: start_thread (pthread_create.c:442)
==38938==    by 0x49A3A03: clone (clone.S:100)
==38938== 
==38938== ----------------------------------------------------------------
==38938== 
==38938== Possible data race during read of size 1 at 0x118128 by thread #1
==38938== Locks held: none
==38938==    at 0x485077C: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4850A2A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10D73C: destruir_mutex (auxiliaresMem.c:77)
==38938==    by 0x10C93D: main (main.c:63)
==38938== 
==38938== This conflicts with a previous write of size 4 by thread #4
==38938== Locks held: none
==38938==    at 0x4917A90: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:62)
==38938==    by 0x4917A90: pthread_mutex_unlock@@GLIBC_2.2.5 (pthread_mutex_unlock.c:368)
==38938==    by 0x4851248: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10DF22: gestor_clientes (server.c:126)
==38938==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4912AC2: start_thread (pthread_create.c:442)
==38938==    by 0x49A3A03: clone (clone.S:100)
==38938==  Address 0x118128 is 8 bytes inside data symbol "mutex_estado_memoria"
==38938== 
==38938== ----------------------------------------------------------------
==38938== 
==38938== Possible data race during read of size 1 at 0x1181A8 by thread #1
==38938== Locks held: none
==38938==    at 0x485077C: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4850A2A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10D75A: destruir_mutex (auxiliaresMem.c:79)
==38938==    by 0x10C93D: main (main.c:63)
==38938== 
==38938== This conflicts with a previous write of size 4 by thread #8
==38938== Locks held: none
==38938==    at 0x4917A90: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:62)
==38938==    by 0x4917A90: pthread_mutex_unlock@@GLIBC_2.2.5 (pthread_mutex_unlock.c:368)
==38938==    by 0x4851248: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10F32D: recibir_cpu (commCPU.c:253)
==38938==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x4912AC2: start_thread (pthread_create.c:442)
==38938==    by 0x49A3A03: clone (clone.S:100)
==38938==  Address 0x1181a8 is 8 bytes inside data symbol "mutex_logs"
==38938== 
==38938== ----------------------------------------------------------------
==38938== 
==38938== Thread #1's call to pthread_mutex_lock failed
==38938==    with error code 22 (EINVAL: Invalid argument)
==38938==    at 0x4850D30: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10C9BC: main (main.c:74)
==38938== 
==38938== ----------------------------------------------------------------
==38938== 
==38938== Thread #1 unlocked an invalid lock at 0x1181A0
==38938==    at 0x48511E6: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10C9E9: main (main.c:76)
==38938== 
==38938== ----------------------------------------------------------------
==38938== 
==38938== Thread #1's call to pthread_mutex_unlock failed
==38938==    with error code 22 (EINVAL: Invalid argument)
==38938==    at 0x4851311: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==38938==    by 0x10C9E9: main (main.c:76)
==38938== 
==38938== 
==38938== Use --history-level=approx or =none to gain increased speed, at
==38938== the cost of reduced accuracy of conflicting-access information
==38938== For lists of detected and suppressed errors, rerun with: -s
==38938== ERROR SUMMARY: 96743 errors from 8 contexts (suppressed: 636521 from 599)
